_data-prepper/pipelines/configuration/processors/otel-trace-group-processor.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: default
+title: OTel trace group processor
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# OpenTelemetry (OTel) trace group processor
+
+The `OTel trace group` processor completes missing trace group related fields in the collection of [span](https://github.com/opensearch-project/data-prepper/blob/834f28fdf1df6d42a6666e91e6407474b88e7ec6/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/trace/Span.java) records by looking up the OpenSearch backend. The `OTel trace group` processor iedntifies the missing trace group information for a `spanId` by looking up the relevant fields in its root `span` stored in OpenSearch, or in the Amazon OpenSearch Service backend that the local Data Prepper host ingests into.
+
+<!--- Do we want to keep "ingests into"?--->
+
+## Usage
+
+<!---Need an introduction here,--->
+
+### OpenSearch
+
+<!---Need an introduction here,--->
+
+```
+pipeline:
+  ...
+  processor:
+    - otel_trace_group:
+        hosts: ["https://localhost:9200"]
+        cert: path/to/cert
+        username: YOUR_USERNAME_HERE
+        password: YOUR_PASSWORD_HERE
+```
+
+See [OpenSearch Security](https://github.com/opensearch-project/data-prepper/blob/834f28fdf1df6d42a6666e91e6407474b88e7ec6/data-prepper-plugins/opensearch/opensearch_security.md#L4) for a detailed explanation. <!--- ...explaination of?--->
+
+### Amazon OpenSearch Service
+
+<!--- Needs an introduction.--->
+
+```
+pipeline:
+  ...
+  processor:
+    - otel_trace_group:
+        hosts: ["https://your-amazon-opensearch-service-endpoint"]
+        aws_sigv4: true
+        cert: path/to/cert
+        insecure: false
+```
+
+See [OpenSearch Sink Security](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/security.md) for a detailed explanation <!---...explanation of?--->.
+
+## Configuration
+
+<!---Need an introduction here,--->
+
+- `hosts`: A list of IP addresses of OpenSearch nodes.
+
+- `cert`(optional): certificate authority (CA) certificate that is pem encoded. Accepts both .pem or .crt. This enables the client to trust the CA that has signed the certificate that OpenSearch is using.
+Default is null.
+
+- `aws_sigv4`: A boolean flag to sign the HTTP request with AWS credentials. Only applies to Amazon OpenSearch Service. See [security](security.md) for details. Default value is `false`.
+
+<!--- Need to fix this link to the security page.--->
```

Comment:
What is the correct link here (security.md)?

Reply:
this needs to be here before we do doc review